core/src/main/scala/filodb.core/memstore/IndexMetadataStore.scala

```diff
@@ -106,4 +110,38 @@ class EphemeralIndexMetadataStore extends IndexMetadataStore {
     initStateMap((datasetRef, shard)) = (state, Some(time))
 }
 
+/**
+ * Non thread safe in memory index state metadata store
+ */
+class FileCheckpointedIndexMetadataStore(val indexLocation: String) extends IndexMetadataStore {
```

Comment:
What is the point of storing the timestamp on file and state in memory? Either both be stored on disk or both in memory. On restart how would you know the state this time in file corresponds to?